### PR TITLE
Feature/multiple stores

### LINKED ISSUE #662: Allow multiple data store

### PR DESCRIPTION
**Pull request by xurxodev**
### :pushpin: References

* **Issue:** Closes #662 

### :memo: Implementation

- I have created a new migration (version 5) to allow multiple stores, this migration creates a new stores key where the value is an array with the current store if exists. I have added to the current store id, default (=true) fields.
- I have created a new store list page
- I have created delete store action. This action is a logic delete, assigning delete=true field to the store.
- I have created a set store as the default action, only one store may be the default.
- I have created a test connection action.
- I have renamed StoreConfigPage to StoreCreationPage to follow the same naming for all creation pages. I have modified this page to use in new and edit mode.
- I have modified publish to store to use default store
- I have modified store validations to not allow to create empty stores
- I have removed the reset button on the store page
- I have modified IntanceSelectionDropdown to can select stores in the packages list page and components where it's used it

- I have performed the minimum tasks that I was clear about this issue due to reduced time to the release and to avoid convert this PR in a very big PR.

Tasks to realize in the future or for this release but in other PR:
- Add name field to stores
- Add a search filter
- Remove Store interface and create a domain entity class 

### :art: Screenshots

Captura de pantalla 2020-10-23 a las 7 32 31

Captura de pantalla 2020-10-23 a las 12 51 00

### :fire: Is there anything the reviewer should know to test it?

In dev.eyeseetea.com:8086 the migration is applied, to test migration in this instance simply create the store key again, remove the stores key, and set version:4 in the config key in the data store.

### :bookmark_tabs: Others

- Any change in the [GUI library](https://github.com/EyeSeeTea/d2-ui-components)? If so, what branch/PR?
-   Any change in the [D2 Api](https://github.com/EyeSeeTea/d2-api)? If so, what branch/PR?

### CODE REVIEW

**Comment by EyeSeeTeaBotTest** [comment]
<h3>Tests failed with errors :rotating_light:</h3><br />Build commit Merge b4992e39a989bc23edc9aad005e4f7bd58f44800 into 6a8716191f7c4766cd2cbb87a0fc15bfc0fd9a1f with commit id 4ae2bcf4e76466c564c6002709d674f342ce0e4f on branch feature/multiple_stores<br /><br />Travis log available at <a href=https://travis-ci.org/EyeSeeTea/metadata-synchronization/builds/738221185 target=_blank>builds/738221185</a><br /><br />See cypress results at <a href=https://eyeseetea.github.io/feedback-ci/EyeSeeTea/metadata-synchronization/pr/666/4ae2bcf4e76466c564c6002709d674f342ce0e4f target=_blank>feedback-ci/EyeSeeTea/metadata-synchronization/pr/666/4ae2bcf4e76466c564c6002709d674f342ce0e4f</a>

**Comment by adrianq** [comment]
I have tested the migration in 8084 which is a snapshot of the WHO with no data from a few weeks ago. Migration works like a charm but it take forever mainly due to backuping hundreds of history records. We will need to take care of this eventually.

Regarding the feature, the only thing I found is a corner-case bug: if I delete the default repository and try to push the package, the app still keeps using the deleted repository. I guess because of the virtual delete. Can we simply do not allow deleting a repo if it is the default repo and show a snackbar accordingly?

File: src/presentation/webapp/pages/store-list/StoreListPage.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        { name: "id", text: i18n.t("Id"), sortable: true, hidden: true },
+        { name: "account", text: i18n.t("Account"), sortable: true },
+        { name: "repository", text: i18n.t("Repository"), sortable: true },
+        { name: "token", text: i18n.t("Token"), sortable: true, hidden: true },
+        {
+            name: "default",
+            text: i18n.t("Default"),
+            sortable: true,
+            getValue: store => {
+                return <Checkbox disabled={true} checked={store.default} />;
```
Comment (SferaDev) [inline src/presentation/webapp/pages/store-list/StoreListPage.tsx:125]: image

```suggestion
                return store.default ? <Icon>done</Icon> : "";
```

Comment (xurxodev) [reply src/presentation/webapp/pages/store-list/StoreListPage.tsx]: thanks @SferaDev

**Comment by xurxodev** [comment]
@adrianq 
- fixed the bug filtering by default and not deleted to publish action. 
- I have modified the behavior to set as default the first store.
- I have modified the render default column as a done icon.

**Pull request merged**